core/src/test/java/org/elasticsearch/script/ScriptServiceTests.java

```diff
@@ -148,6 +148,39 @@ public void testScriptsWithoutExtensions() throws IOException {
         }
     }
 
+    public void testHiddenFileSkipped() throws IOException {
+        ContextAndHeaderHolder contextAndHeaders = new ContextAndHeaderHolder();
+        buildScriptService(Settings.EMPTY);
+
+        logger.info("--> setup one hidden test file");
+        Path testFileHidden = scriptsFilePath.resolve(".hidden_file");
+        Path testRegularFile = scriptsFilePath.resolve("test_file.tst");
+        Streams.copy("test_hidden_file".getBytes("UTF-8"), Files.newOutputStream(testFileHidden));
+        Streams.copy("test_file".getBytes("UTF-8"), Files.newOutputStream(testRegularFile));
+        resourceWatcherService.notifyNow();
+
+        try {
+            logger.info("--> verify if hidden_file was skipped");
+            scriptService.compile(new Script("hidden_file", ScriptType.FILE, "test", null),
```

Comment:
I don't think so, that's relying too much on an internal implementation detail (that there is a cache, that file scripts are compiled and put into the static cache, that the key is the prefix of the filename, etc.). The purpose behind the PR is to get the script service to ignore hidden files, and that is what needs to be tested. I haven't looked too closely, but I suspect that you're going to have to hook into the script service or maybe the resource watcher and possibly refactor a little bit to expose the pieces needed to in fact make this assertion. Let me know if that's enough to get you started, I'm happy to take a closer look if needed. :)